### PR TITLE
fix(ci): Secrets unavailable for deploy jobs

### PR DESCRIPTION
**Pull request by khvn26**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Fixes deploy jobs. 

## How did you test this code?

This is a CI change.

### CODE REVIEW

File: .github/workflows/.reusable-deploy-ecs.yml [hunk 1]
```diff
@@ -61,7 +62,7 @@ jobs:
           build-args: CI_COMMIT_SHA=${{ github.sha }}
           secrets: |
             github_private_cloud_token=${{ secrets.GH_PRIVATE_ACCESS_TOKEN }}
-            sse_pgp_pkey=${{ secrets.SSE_PGP_PRIVATE_KEY }}
+            "sse_pgp_pkey=${{ secrets.SSE_PGP_PRIVATE_KEY }}"
```
Comment (matthewelwell) [inline .github/workflows/.reusable-deploy-ecs.yml:65]: Odd that it's only this one that needs quotations. Could / should we add a comment?

File: Dockerfile [hunk 2]
```diff
@@ -167,8 +167,7 @@ FROM api-runtime-private as saas-api
 RUN --mount=type=secret,id=sse_pgp_pkey \
   apt-get update && apt-get install -y gnupg && \
   gpg --import /run/secrets/sse_pgp_pkey && \
-  mv /root/.gnupg /app/; \
-  chown -R nobody /app/.gnupg
+  mv /root/.gnupg/ /app/
```
Comment (matthewelwell) [inline Dockerfile]: I'm keen to get @gagantrivedi's review here as well to make sure we know what we need to test.

Comment (khvn26) [reply Dockerfile]: Sure. The image has been deployed to staging so we just need to verify SSE there.

Comment (gagantrivedi) [reply Dockerfile]: Yeah, we have to make sure `sse` events are still reaching influx

File: .github/workflows/.reusable-deploy-ecs.yml (hunk 1 above)
Comment (khvn26) [reply .github/workflows/.reusable-deploy-ecs.yml:65]: The reason for the quotes is that the value is multiline.

Comment (khvn26) [reply .github/workflows/.reusable-deploy-ecs.yml:65]: A comment would be tough because yaml `|` makes it impossible to comment the same line.

Comment (matthewelwell) [reply .github/workflows/.reusable-deploy-ecs.yml:65]: Ok, yep, fair point. Hopefully this conversation will serve enough if people are searching for a reason in the future...

File: Dockerfile [hunk 3]
```diff
@@ -167,8 +167,8 @@ FROM api-runtime-private as saas-api
 RUN --mount=type=secret,id=sse_pgp_pkey \
   apt-get update && apt-get install -y gnupg && \
   gpg --import /run/secrets/sse_pgp_pkey && \
-  mv /root/.gnupg /app/; \
-  chown -R nobody /app/.gnupg
+  mv /root/.gnupg/ /app/ && \
```
Comment (gagantrivedi) [inline Dockerfile:170]: Are you able to build this image without any GPG key?

Comment (khvn26) [reply Dockerfile:170]: > Are you able to build this image without any GPG key?

There's no need as `saas-api` target is only intended for building with the key.

Comment (gagantrivedi) [reply Dockerfile:170]: > > Are you able to build this image without any GPG key?
> 
> There's no need as `saas-api` target is only intended for building with the key.

Hmm, I am little confused… Is this file not used for private cloud? What if we want to build an image locally to test something?

Comment (khvn26) [reply Dockerfile:170]: This file is used for all of our targets now. To build the `private-cloud` target, you don't need the key; the GPG import is the only layer added by the `saas-api` target stage.

I'd love to make the key a runtime dependency and shave off one target, though! Let me know if you have an idea how to do that.

See Dockerfile comments and the [PR description](https://github.com/Flagsmith/flagsmith/pull/4165#issue-2353549270) for details.

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**